### ISSUE TITLE
Option to disable bracketed paste (since it breaks Python REPL)

### ISSUE THREAD

**Issue by svet-b** [fish-shell/fish-shell#7906]
Due to a regression, pasting in the Python interpreter is broken in some recent versions - as documented in e.g. https://github.com/Homebrew/homebrew-core/issues/68193. The commonly recommended workaround is to
```
set enable-bracketed-paste off
```
which is obviously bash/zsh-specific. According to https://github.com/fish-shell/fish-shell/pull/3871#issuecomment-321034918 it seems that bracketed paste can in fact not be turned off in fish. An option to turn it off would be desirable.

#### Context, if relevant:
Please tell us which fish version you are using: 3.1.2
Please tell us which operating system and terminal you are using: macOS 11.2.1/ARM64, with built-in Terminal app.
No third party customizations.
Running Python 3.8.3 installed via Homebrew.

**Comment by faho** [comment]
Fish should already turn off bracketed paste while anything else is running, so I'm not entirely sure what's happening here.

Is this actually an issue with fish? The workaround isn't to turn it off in bash, it's to turn it off in *readline*, which is also what the python REPL uses.

(also obviously it would be preferable if the underlying bug was just fixed)

**Comment by svet-b** [comment]
Thanks @faho for the quick response! Indeed, it would be great if the upstream issue was fixed!

> Is this actually an issue with fish? The workaround isn't to turn it off in bash, it's to turn it off in readline, which is what both bash and the python REPL use as their input library. So exactly that workaround also applies if you are running fish as your shell, because it's not about turning it off in your shell.

Thanks for clarifying this, I hadn't realized that it's more Python-related than shell-related. I'm reasonably computer-literate, though only came across "bracketed paste" this morning.

As a quick test, I actually tried the workaround in bash and also ended up with the same issue, so it looks like I'll need to keep digging. Happy to close this as it's maybe not relevant after all.
